### PR TITLE
Fix bug on create OCI Manifest

### PR DESCRIPTION
**Pull request by clarelimo**
# Unable to create OCI manifest in Azure Container Registry	

## Issue Statement

The SDK doesn't enable the client to create an OCI manifest.

This issue is tracked at [Azure/azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net/issues/17084).

## Root Cause Analysis

The MediaType used in the http request headers(content type) was that of a V2Manifest leaving out all the other MediaTypes for other types of manifests.

## Resolution

To fix the issue, the media type was added as a parameter in each of the manifest types and all of them were fetched in the _httpRequest.Content.Headers.ContentType.

## Tests
Automated tests were added to create the different manifests and they were all successful.

### CODE REVIEW

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Generated/ManifestsOperations.cs [hunk 1]
```diff
@@ -358,10 +358,30 @@ public async Task<AzureOperationResponse<ManifestWrapper>> GetWithHttpMessagesAs
             // Serialize Request
             string _requestContent = null;
             if(payload != null)
-            {
+            {
```
Comment (avtakkar) [inline sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Generated/ManifestsOperations.cs:361]: Nit: remove whitespace

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Generated/ManifestsOperations.cs [hunk 2]
```diff
@@ -358,10 +358,30 @@ public async Task<AzureOperationResponse<ManifestWrapper>> GetWithHttpMessagesAs
...
+                        break;
+                    case OCIManifest ociManifest:
+                        _httpRequest.Content.Headers.ContentType = System.Net.Http.Headers.MediaTypeHeaderValue.Parse(ociManifest.MediaType);
+                        break;
+                    case OCIIndex ociIndex:
+                        _httpRequest.Content.Headers.ContentType = System.Net.Http.Headers.MediaTypeHeaderValue.Parse(ociIndex.MediaType);
+                        break;
+                    default:
+                        throw new AcrErrorsException($"The requested mediaType could not be found in {payload}.");
+                }
```
Comment (avtakkar) [inline sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Generated/ManifestsOperations.cs:384]: You could refactor this as:
```csharp
string mediaType = string.Empty;
switch (payload)
{
  case OCIIndex ociIndex:
      mediaType = ociIndex.MediaType;
  ...                     
}
MediaTypeHeaderValue.Parse(mediaType);
```

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Generated/ManifestsOperations.cs [hunk 3]
```diff
@@ -358,10 +358,30 @@ public async Task<AzureOperationResponse<ManifestWrapper>> GetWithHttpMessagesAs
...
+                        _httpRequest.Content.Headers.ContentType = System.Net.Http.Headers.MediaTypeHeaderValue.Parse(manifestList.MediaType);
+                        break;
+                    case OCIManifest ociManifest:
+                        _httpRequest.Content.Headers.ContentType = System.Net.Http.Headers.MediaTypeHeaderValue.Parse(ociManifest.MediaType);
+                        break;
+                    case OCIIndex ociIndex:
+                        _httpRequest.Content.Headers.ContentType = System.Net.Http.Headers.MediaTypeHeaderValue.Parse(ociIndex.MediaType);
+                        break;
+                    default:
+                        throw new AcrErrorsException($"The requested mediaType could not be found in {payload}.");
```
Comment (avtakkar) [inline sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Generated/ManifestsOperations.cs:383]: In this case, the payload is not a supported type at all. So maybe the message should say "unknown payload type" or something along those lines.

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs [hunk 4]
```diff
@@ -78,6 +78,7 @@ public class ManifestTests
         private static readonly V1Manifest ExpectedV1ManifestProd = new V1Manifest()
         {
             SchemaVersion = 1,
+            MediaType=ACRTestUtil.MediatypeV1Manifest,
```
Comment (avtakkar) [inline sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs:81]: Nit: formatting

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs [hunk 5]
```diff
@@ -191,6 +192,7 @@ public class ManifestTests
 
         private static readonly OCIManifest ExpectedOCIManifestProd = new OCIManifest()
         {
+            MediaType=ACRTestUtil.MediatypeOCIManifest,
```
Comment (avtakkar) [inline sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs:195]: nit: formatting

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs [hunk 6]
```diff
@@ -515,7 +557,7 @@ private void VerifyManifest(Manifest baseManifest, Manifest actualManifest)
         {
             Assert.Equal(baseManifest.GetType(), actualManifest.GetType());
             Assert.Equal(baseManifest.SchemaVersion, actualManifest.SchemaVersion);
-
+
```
Comment (avtakkar) [inline sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs:560]: remove whitespace

**Review by yugangw-msft — APPROVED** [review]

**Pull request merged**